tpu-client/src/nonblocking/tpu_client.rs

```diff
@@ -272,6 +272,44 @@ pub struct TpuClient<
     connection_cache: Arc<ConnectionCache<P, M, C>>,
 }
 
+async fn try_send_wire_transaction<P, M, C>(
+    wire_transaction: Vec<u8>,
+    leaders: &[SocketAddr],
+    connection_cache: &ConnectionCache<P, M, C>,
+) -> TransportResult<()>
+where
+    P: ConnectionPool<NewConnectionConfig = C>,
+    M: ConnectionManager<ConnectionPool = P, NewConnectionConfig = C>,
+    C: Send + Sync + 'static,
+{
+    let futures = leaders
+        .iter()
+        .map(|addr| send_wire_transaction_to_addr(connection_cache, addr, wire_transaction.clone()))
+        .collect::<Vec<_>>();
+    let results: Vec<TransportResult<()>> = join_all(futures).await;
+
+    let mut last_error: Option<TransportError> = None;
+    let mut some_success = false;
+    for result in results {
+        if let Err(e) = result {
+            if last_error.is_none() {
+                last_error = Some(e);
+            }
+        } else {
+            some_success = true;
+        }
+    }
+    if !some_success {
+        Err(if let Some(err) = last_error {
+            err
+        } else {
+            std::io::Error::new(std::io::ErrorKind::Other, "No sends attempted").into()
+        })
+    } else {
+        Ok(())
+    }
```

Comment:
i'm finding this logic a bit convoluted, how about...

```suggestion
    let any_attempted = !results.is_empty();
    let any_succeeded = results.any(|result| result.is_ok());
    let result = results.into_iter().collect::<TransportResult<Vec<()>>();
    if any_succeeded {
        Ok(())
    } else !any_attempted {
        Err(std::io::Error::new(std::io::ErrorKind::Other, "No sends attempted").into())
    } else {
        result.map(|_| ())
    }
```

Reply:
derp... in a follow up i guess. since this is just moved from elsewhere

Reply:
Yeah I just wanted to refactor that logic and not change it